src/web/server.py

```
@@ -0,0 +1,586 @@
+                        os.makedirs(self._photo_directory, exist_ok=True)
+
+                        # Create safe filename
+                        import datetime
+                        filename = f"Photo_{datetime.datetime.utcnow().strftime('%Y%m%d_%H%M%S')}.jpg"
+                        filepath = os.path.join(self._photo_directory, filename)
+
+                        # Validate path
+                        base_dir = os.path.abspath(self._photo_directory)
+                        abs_path = os.path.abspath(filepath)
+                        if os.path.commonpath([abs_path, base_dir]) != base_dir:
+                            raise ValueError("Invalid photo path")
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)